analyzers/src/SonarAnalyzer.CSharp/Rules/RedundantModifier.cs

```diff
@@ -379,40 +326,35 @@ private void VisitChecked<T>(T node, SyntaxKind checkedKind, SyntaxToken tokenTo
                 where T : SyntaxNode
             {
                 var isThisNodeChecked = node.IsKind(checkedKind);
-
                 var originalIsCurrentContextChecked = isCurrentContextChecked;
                 var originalContextHasIntegralOperation = currentContextHasIntegralOperation;
-
                 isCurrentContextChecked = isThisNodeChecked;
                 currentContextHasIntegralOperation = false;
 
                 baseCall(node);
 
                 var isSimplyRedundant = IsCurrentNodeEmbeddedInsideSameChecked(node, isThisNodeChecked, originalIsCurrentContextChecked);
-
                 if (isSimplyRedundant || !currentContextHasIntegralOperation)
                 {
                     var keywordToReport = isThisNodeChecked ? "checked" : "unchecked";
                     context.ReportIssue(Diagnostic.Create(Rule, tokenToReport.GetLocation(), keywordToReport, "redundant"));
                 }
-
                 isCurrentContextChecked = originalIsCurrentContextChecked;
                 currentContextHasIntegralOperation = originalContextHasIntegralOperation || (currentContextHasIntegralOperation && isSimplyRedundant);
             }
 
             private bool IsCurrentNodeEmbeddedInsideSameChecked(SyntaxNode node, bool isThisNodeChecked, bool isCurrentContextChecked) =>
-                node != context.Node && isThisNodeChecked == isCurrentContextChecked;
+                isThisNodeChecked == isCurrentContextChecked && node != context.Node;
 
             private void SetHasIntegralOperation(CastExpressionSyntax node)
             {
                 var expressionType = context.SemanticModel.GetTypeInfo(node.Expression).Type;
                 var castedToType = context.SemanticModel.GetTypeInfo(node.Type).Type;
-                currentContextHasIntegralOperation |= castedToType != null && expressionType != null && castedToType.IsAny(KnownType.IntegralNumbers);
+                currentContextHasIntegralOperation |= castedToType is not null && expressionType is not null && castedToType.IsAny(KnownType.IntegralNumbers);
             }
 
             private void SetHasIntegralOperation(ExpressionSyntax node) =>
-                currentContextHasIntegralOperation |= context.SemanticModel.GetSymbolInfo(node).Symbol is IMethodSymbol methodSymbol
-                                                      && methodSymbol.ReceiverType.IsAny(KnownType.IntegralNumbers);
+                currentContextHasIntegralOperation |= context.SemanticModel.GetSymbolInfo(node).Symbol is IMethodSymbol methodSymbol && methodSymbol.ReceiverType.IsAny(KnownType.IntegralNumbers);
```

Comment:
Is `|=` short-circuiting? Otherwise `context.SemanticModel.GetSymbolInfo(node)` might be called for no reason.

Reply:
No, it's not.